libs/vscode-ui/components/src/lib/autocomplete/autocomplete.component.ts

```diff
@@ -1,16 +1,225 @@
-import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
+import {
+  Component,
+  OnDestroy,
+  Input,
+  SimpleChanges,
+  OnChanges,
+  ViewChild,
+  ElementRef,
+  OnInit,
+  ViewEncapsulation
+} from '@angular/core';
+import {
+  ControlValueAccessor,
+  NG_VALUE_ACCESSOR,
+  FormControl
+} from '@angular/forms';
+import {
+  map,
+  tap,
+  takeUntil,
+  switchMap,
+  startWith,
+  scan,
+  debounceTime,
+  filter,
+  shareReplay
+} from 'rxjs/operators';
+import {
+  Subject,
+  BehaviorSubject,
+  Observable,
+  fromEvent,
+  merge,
+  of
+} from 'rxjs';
 import { Schema } from '@angular-console/schema';
 
-/**
- * Not built yet - using a datalist temporarily to loosely mimic functionality
- */
+enum AutocompleteNavKeys {
+  Enter = 'Enter',
+  ArrowUp = 'ArrowUp',
+  ArrowDown = 'ArrowDown'
+}
 
 @Component({
   selector: 'angular-console-autocomplete',
   templateUrl: './autocomplete.component.html',
   styleUrls: ['./autocomplete.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush
+  encapsulation: ViewEncapsulation.None,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: AutocompleteComponent,
+      multi: true
+    }
+  ]
 })
-export class AutocompleteComponent {
+export class AutocompleteComponent
+  implements OnInit, OnDestroy, OnChanges, ControlValueAccessor {
   @Input() field: Schema;
+
+  @Input() options: string[] = [];
+  private readonly _options$ = new BehaviorSubject<string[]>(this.options);
+  visibleOptions: Observable<string[]>;
+  selectedOption$: Observable<number | null>;
+
+  @ViewChild('textInput', { static: true }) textInput: ElementRef;
+  control: FormControl;
+  isFocused$: Observable<boolean>;
+
+  private readonly _destroying$ = new Subject<void>();
```

Comment:
The rest of the code base doesn't use _ prefixing for private variables nor do we use the $ suffix for subjects. Consider sticking to our current conventions for consistency

Reply:
Renamed to `destroying`